source/ada/lsp-ada_handlers.adb

```diff
@@ -2647,81 +2648,96 @@ package body LSP.Ada_Handlers is
       Document  : constant LSP.Ada_Documents.Document_Access :=
                    Get_Open_Document (Self, Value.textDocument.uri);
 
+      Refs      : Laltools.Refactor.Rename.Renamable_References;
+
       procedure Process_Context (C : Context_Access);
       --  Process the rename request for the given context, and add
-      --  the results to response.
+      --  the results to response. If a rename problem is found, return a
```

Comment:
(clarification): this says "if a problem is found, return a response with an error": at first I thought this should be transformed into a function that actually returns the error... I suggest to clarify this by explaining which mechanism is being used to return the error to the user.